### PR TITLE
Use tools/print_test_stats.py in GHA

### PR DESCRIPTION
**Pull request by samestep**
Judging from #57584, it seems like the test-reports artifact was originally intended to be downloaded to `$PWD/test-reports` instead of just directly into `$PWD`. To minimize confusion, this PR changes it to download into `test/test-reports`, which should match where the files came from in the `test` step anyway.

TODOs:

- [x] change the extract path for test-reports
- [x] install Python dependencies
- [x] call `tools/print_test_stats.py`
- [x] use deep clone to allow `git` commands
- [x] correctly set `CIRCLE_*` environment variables
- [x] set Scribe credentials
- [x] set AWS credentials

**Test plan:**

CI.

### CODE REVIEW

File: .github/templates/linux_ci_workflow.yml.in [hunk 1]
```diff
@@ -208,8 +208,12 @@ jobs:
...
+        # boto3 version copied from .circleci/docker/common/install_conda.sh
         run: |
-          pip install junitparser rich
+          pip install -r requirements.txt
+          pip install boto3==1.16.34 junitparser rich
       - name: Output Test Results (Click Me)
         run: |
           python tools/render_junit.py .
+      - name: Display and upload test statistics (Click Me)
+        run: python tools/print_test_stats.py --upload-to-s3 --compare-with-s3 test
```
Comment (walterddr) [inline .github/templates/linux_ci_workflow.yml.in:219]: export PYTHONPATH="\${PWD}"

File: .github/workflows/pytorch-linux-xenial-py3.6-gcc5.4.yml [hunk 2]
```diff
@@ -201,16 +201,41 @@ jobs:
...
+        env:
+          SCRIBE_GRAPHQL_ACCESS_TOKEN: ${{ secrets.SCRIBE_GRAPHQL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_S3_UPDATE_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_S3_UPDATE_SECRET_ACCESS_KEY }}
+          CIRCLE_BRANCH: ${{ steps.parse-ref.outputs.branch }}
+          CIRCLE_JOB: pytorch-linux-xenial-py3.6-gcc5.4
+          CIRCLE_PR_NUMBER: ${{ github.event.pull_request.number }}
+          CIRCLE_SHA1: ${{ github.event.pull_request.head.sha || github.sha }}
+          CIRCLE_TAG: ${{ steps.parse-ref.outputs.tag }}
+          CIRCLE_WORKFLOW_ID: ${{ github.run_id }} # dunno if this corresponds
```
Comment (janeyx99) [inline .github/workflows/pytorch-linux-xenial-py3.6-gcc5.4.yml:238]: What do we use this for?

Comment (samestep) [reply .github/workflows/pytorch-linux-xenial-py3.6-gcc5.4.yml:238]: dunno; @walterddr do you happen to know?

File: .github/templates/linux_ci_workflow.yml.in [hunk 3]
```diff
@@ -200,16 +200,41 @@ jobs:
...
+          python tools/render_junit.py test
+      - name: Parse ref
+        id: parse-ref
+        run: .github/scripts/parse_ref.py
+      - name: Display and upload test statistics (Click Me)
+        # temporary hack: set CIRCLE_* vars, until we update
+        # tools/print_test_stats.py to natively support GitHub Actions
+        env:
+          SCRIBE_GRAPHQL_ACCESS_TOKEN: ${{ secrets.SCRIBE_GRAPHQL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_S3_UPDATE_ACCESS_KEY_ID }}
```
[the remark this answers is not in the document]
Comment (samestep) [reply .github/templates/linux_ci_workflow.yml.in:230]: Yep, it still works on PRs even though those secrets are absent: https://github.com/pytorch/pytorch/runs/2548268438

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**